vpr/src/place/noc_place_utils.cpp

```diff
@@ -2,46 +2,42 @@
 /*********************************************************** *****************************/
 
 void initial_noc_placement(void) {
-    // need to get placement information about where the router cluster blocks are palced on the device
+    // need to get placement information about where the router cluster blocks are placed on the device
     const auto& place_ctx = g_vpr_ctx.placement();
 
     // need to update the link usages within after routing all the traffic flows
-    // also need to route all the traffic flows ans store them
+    // also need to route all the traffic flows and store them
     auto& noc_ctx = g_vpr_ctx.mutable_noc();
 
     NocTrafficFlows* noc_traffic_flows_storage = &noc_ctx.noc_traffic_flows_storage;
 
     /* We need all the traffic flow ids to be able to access them. The range
      * of traffic flow ids go from 0 to the total number of traffic flows within
-     * the NoC. So get the upper range here.*/
-    int number_of_traffic_flows = noc_traffic_flows_storage->get_number_of_traffic_flows();
-
-    // go through all the traffic flows and route them. Then once routed, update the links used in the routed traffic flows with their usages
-    for (int traffic_flow_id = 0; traffic_flow_id < number_of_traffic_flows; traffic_flow_id++) {
-        // get the traffic flow with the current id
-        NocTrafficFlowId conv_traffic_flow_id = (NocTrafficFlowId)traffic_flow_id;
-        const t_noc_traffic_flow& curr_traffic_flow = noc_traffic_flows_storage->get_single_noc_traffic_flow(conv_traffic_flow_id);
+     * the NoC.
+     * go through all the traffic flows and route them. Then once routed, update the links used in the routed traffic flows with their usages
+     */
+    for (const auto& traffic_flow_id : noc_traffic_flows_storage->get_all_traffic_flow_id()) {
+        const t_noc_traffic_flow& curr_traffic_flow = noc_traffic_flows_storage->get_single_noc_traffic_flow(traffic_flow_id);
 
         // update the traffic flow route based on where the router cluster blocks are placed
-        std::vector<NocLinkId>& curr_traffic_flow_route = get_traffic_flow_route(conv_traffic_flow_id, noc_ctx.noc_model, *noc_traffic_flows_storage, noc_ctx.noc_flows_router, place_ctx.block_locs);
+        std::vector<NocLinkId>& curr_traffic_flow_route = get_traffic_flow_route(traffic_flow_id, noc_ctx.noc_model, *noc_traffic_flows_storage, *noc_ctx.noc_flows_router, place_ctx.block_locs);
 
-        // update the links used in the found traffic flow route
-        update_traffic_flow_link_usage(curr_traffic_flow_route, noc_ctx.noc_model, link_usage_update_state::increment, curr_traffic_flow.traffic_flow_bandwidth);
+        // update the links used in the found traffic flow route, links' bandwidth should be incremented since the traffic flow is routed
+        update_traffic_flow_link_usage(curr_traffic_flow_route, noc_ctx.noc_model, 1, curr_traffic_flow.traffic_flow_bandwidth);
```

Comment:
Could still use the enumerated constant, but just set the values to -1 (decrement) and +1 (increment). Check what looks nicer in the code.